### PR TITLE
include last parent in fetch list of L1s for transplant

### PR DESCRIPTION
**Pull request by rymurr**
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/434)

### CODE REVIEW

File: versioned/jgit/src/test/java/com/dremio/nessie/versioned/impl/ITJGitInMemoryVersionStore.java [hunk 1]
```diff
@@ -78,4 +78,7 @@ void tearDown() {
   protected void transplant() {
   }
 
+  @Disabled
+  protected void transplantBasic() {
+  }
```
Comment (rymurr) [inline versioned/jgit/src/test/java/com/dremio/nessie/versioned/impl/ITJGitInMemoryVersionStore.java:83]: temporary till #430 is merged

File: versioned/jgit/src/test/java/com/dremio/nessie/versioned/impl/ITJGitOnDiskVersionStore.java [hunk 2]
```diff
@@ -82,4 +82,7 @@ void tearDown() {
   protected void transplant() {
   }
 
+  @Disabled
```
Comment (rymurr) [inline versioned/jgit/src/test/java/com/dremio/nessie/versioned/impl/ITJGitOnDiskVersionStore.java:85]: temporary till #430 is merged

File: versioned/dynamodb/src/main/java/com/dremio/nessie/versioned/impl/TieredVersionStore.java [hunk 3]
```diff
@@ -465,10 +469,9 @@ public void transplant(BranchName targetBranch, Optional<Hash> currentBranchHash
         true,
         (from, commonParent) -> {
           // first we need to validate that the actual history matches the provided sequence.
-          List<L1> l1s = Lists.reverse(
-              new HistoryRetriever(store, from, endTarget, true, false, true).getStream().map(HistoryItem::getL1)
-              .collect(ImmutableList.toImmutableList()));
-          List<Hash> hashes = l1s.stream().map(L1::getId).map(Id::toHash).collect(Collectors.toList());
+          Stream<L1> historyStream = new HistoryRetriever(store, from, null, true, false, true).getStream().map(HistoryItem::getL1);
+          List<L1> l1s = Lists.reverse(takeUntilNext(historyStream, endTarget).collect(ImmutableList.toImmutableList()));
+          List<Hash> hashes = l1s.stream().map(L1::getId).map(Id::toHash).skip(1).collect(Collectors.toList());
```
Comment (jacques-n) [inline versioned/dynamodb/src/main/java/com/dremio/nessie/versioned/impl/TieredVersionStore.java:474]: why are you skipping 1?

**Comment by rymurr** [comment]
1 is always going to be the commit prior to the first cherry pick. So 2..n
will match the requested array.

**Comment by jacques-n** [comment]
I'm not following why you would skip the first commit. If I say grab c1, you'd get a history starting at c1 and ending at c1 - 1. Why would you drop c1?

**Comment by rymurr** [comment]
> I'm not following why you would skip the first (most recent) commit. If I say grab c1, you'd get a history starting at c1 and ending at c1 - 1. Why would you drop c1?

Correct, we ask for `c1` and the list comes back as `{c1, c1-1}` which is then reversed and we drop `c1-1` so the list is just `c1` to match the assertion. however the list `{c1-1, c1}` is returned back to the `internalTransplant` method.

So we return N+1 `L1`s (where N is the number of commits to transplant) and we also ensure that the last N commits match the commits we wanted to transplant.

**Review by jacques-n — APPROVED** [review]
Got it. My skim didn't clarify that the skip is on the l1 list, not the history retriever.

**Review by jacques-n — APPROVED** [review]

**Pull request merged**